File: frontend/src/types/lake.ts
export type RiskTier = 'critical' | 'high' | 'advisory' | 'safe'

export type Lake = {
  id: string
  nodeId: string
  name: string
  basin: string
  state: string
  lat: number
  lng: number
  riskScore: number
  tier: RiskTier
  vulnerablePop: number
  lastUpdated: string
  watchId: string
  predictions: { h72: number; h24: number; h6: number }
  telemetry: {
    waterLevelMPerHr: number
    seismicMag: number
    areaDeltaPct: number
  }
  shap: { feature: string; value: number; direction: 'up' | 'down' }[]
  floodProjection: { timeBand: string; village: string; popAtRisk: number; urgency: 'critical' | 'elevated' | 'watch' }[]
  escalationLevel: number
  smsSent: number
}

File: backend/schemas/lake.py
from typing import List, Optional, Any, Dict
from pydantic import BaseModel, Field, ConfigDict

class PredictionsSchema(BaseModel):
    h72: float = 0.0
    h24: float = 0.0
    h6: float = 0.0

class TelemetrySchema(BaseModel):
    waterLevelMPerHr: float = 0.0
    seismicMag: float = 0.0
    areaDeltaPct: float = 0.0

class ShapItemSchema(BaseModel):
    feature: str
    value: float
    direction: str  # 'up' | 'down'

class FloodProjectionItemSchema(BaseModel):
    timeBand: str
    village: str
    popAtRisk: int
    urgency: str  # 'critical' | 'elevated' | 'watch'

class LakeResponse(BaseModel):
    model_config = ConfigDict(populate_by_name=True, from_attributes=True)

    id: str
    nodeId: str = Field(alias="node_id")
    name: str
    basin: str
    state: str
    lat: float
    lng: float
    riskScore: float = Field(alias="risk_score")
    tier: str
    vulnerablePop: int = Field(alias="vulnerable_pop")
    lastUpdated: str = Field(alias="last_updated")
    watchId: str = Field(alias="watch_id")
    predictions: PredictionsSchema
    telemetry: TelemetrySchema
    shap: List[ShapItemSchema] = []
    floodProjection: List[FloodProjectionItemSchema] = Field(default=[], alias="flood_projection")
    escalationLevel: int = Field(default=0, alias="escalation_level")
    smsSent: int = Field(default=0, alias="sms_sent")
    weatherData: Optional[Dict[str, Any]] = Field(default=None, alias="weather_data")

class LakeSyncResponse(BaseModel):
    status: str
    lake_id: str
    weather_summary: Dict[str, Any]
    new_risk_score: float
    new_tier: str
